player/src/main/java/com/focusit/jsflight/player/scenario/ScenarioProcessor.java

```diff
@@ -301,13 +300,18 @@ public void play(UserScenario scenario, SeleniumDriver seleniumDriver, int start
         int maxPosition = finish > 0 ? Math.min(finish, scenario.getStepsCount()) : scenario.getStepsCount();
         LOG.info("Playing scenario. Start step: {}, finish step: {}. Steps count: {}", start, maxPosition,
                 maxPosition - start);
-        while (scenario.getPosition() != maxPosition)
-        {
-            LOG.info("Step " + scenario.getPosition());
-            applyStep(scenario, seleniumDriver, scenario.getPosition());
-            scenario.moveToNextStep();
+        try {
+            while (scenario.getPosition() != maxPosition) {
+                LOG.info("Step " + scenario.getPosition());
+                applyStep(scenario, seleniumDriver, scenario.getPosition());
+                scenario.moveToNextStep();
+            }
+            LOG.info(String.format("Done(%d):playing", System.currentTimeMillis() - begin));
+        } catch (Throwable throwable) {
+            LOG.error("Playing interrupted by unhandled exception", throwable);
+            throw throwable;
+        } finally {
+            seleniumDriver.closeWebDrivers();
```

Comment:
As far as I remember, I wanted to leave browsers opened to have an opportunity to continue playing scenario after some problems gets resolved. 
Imagine a scenario passed 950 step of 1000 and the something came up. 
You could fix some issues and you might want to continue playing the scenario started at 950 step of 1000.